src/main/java/com/ibm/stocator/fs/cos/COSAPIClient.java

```diff
@@ -227,7 +228,7 @@
    * Contains map of object names that were created. Used in container
    * listing
    */
-  private Map<String, Boolean> mCachedSparkOriginated;
+  private ConcurrentHashMap<String, Boolean> mCachedSparkOriginated;
```

Comment:
Style nit, declaration should probably remain an interface, either leave as `Map`, or switch to `ConcurrentMap` if any concurrency specific methods are called

Reply:
open new merge request to master as @gilv  ask. https://github.com/CODAIT/stocator/pull/213